### PR TITLE
Update for alpha.1 HIP-18 protos

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Summary of the change**:
- Adopt the protobufs approved for alpha HIP-18 release [here](https://github.com/hashgraph/hedera-protobufs/pull/42/files) via a `0.16.0-alpha.1-SNAPSHOT` artifact.
- Eliminate `customFeesKey`, use `canUpdateWithAdminKey` flag instead.
- Add overflow checks in `ImpliedTransfersMarshal` (resolve to`CUSTOM_FEE_OUTSIDE_NUMERIC_RANGE` in this case).
- Add checks in `HederaTokenStore` for negative fees (resolve to `CUSTOM_FEE_MUST_BE_POSITIVE` in this case).

### CODE REVIEW

File: hedera-node/src/main/java/com/hedera/services/grpc/marshalling/ImpliedTransfersMarshal.java [hunk 1]
```diff
@@ -134,84 +157,127 @@ public ImpliedTransfers unmarshalFromGrpc(CryptoTransferTransactionBody op, Acco
...
-				existingBalanceChanges, customFeeChanges, -feesToCollect,
-				tokenAdjust(payerId, scopingToken, -feesToCollect), true);
+		assessedCustomFeesForRecord.add(
+				new AssessedCustomFee(fees.getFeeCollectorAccountId(), scopingToken.asEntityId(), effectiveFee));
+	}
 
-		assessedCustomFeesForRecord.add(new AssessedCustomFee(fees.getFeeCollector(),
-				scopingToken.asEntityId(),
-				feesToCollect));
+	long safeFractionMultiply(long n, long d, long v) {
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/grpc/marshalling/ImpliedTransfersMarshal.java:196]: Very Nice 😃

File: hedera-node/src/main/java/com/hedera/services/state/submerkle/CustomFee.java [hunk 2]
```diff
@@ -271,8 +268,8 @@ public FractionalFeeSpec(
 			if (denominator == 0) {
 				throw new IllegalArgumentException("Division by zero is not allowed");
 			}
-			this.numerator = numerator;
-			this.denominator = denominator;
+			this.numerator = Math.abs(numerator);
+			this.denominator = Math.abs(denominator);
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/state/submerkle/CustomFee.java:272]: should we not throw an exception when the values provided are negative?

File: hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java [hunk 3]
```diff
@@ -442,14 +443,19 @@ private ResponseCodeEnum validateFeeSchedule(List<CustomFeesOuterClass.CustomFee
...
-				final var fraction = fractionalSpec.getFractionOfUnitsToCollect();
+				final var fraction = fractionalSpec.getFractionalAmount();
 				if (fraction.getDenominator() == 0) {
 					return FRACTION_DIVIDES_BY_ZERO;
 				}
+				if (!signsMatch(fraction.getNumerator(), fraction.getDenominator())) {
+					return CUSTOM_FEE_MUST_BE_POSITIVE;
+				}
+				if (fractionalSpec.getMaximumAmount() < 0 || fractionalSpec.getMinimumAmount() < 0) {
+					return CUSTOM_FEE_MUST_BE_POSITIVE;
```
Comment (Neeharika-Sompalli) [inline hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java:457]: Should there be a check that MaximumAmount > MinimumAmount ?

**Review by anighanta — DISMISSED** [review]
Agree with @Neeharika-Sompalli we might want a new response code if maxFee is less than minFee in the FractionalFee. 
LGTM otherwise.

File: hedera-node/src/main/java/com/hedera/services/state/submerkle/CustomFee.java (hunk 2 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/state/submerkle/CustomFee.java]: Good idea, done!

File: hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java (hunk 3 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/store/tokens/HederaTokenStore.java:457]: Also done!!

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM!

**Review by qnswirlds — DISMISSED** [review]
Overall LGTM. Just a few nits.

**Review by Neeharika-Sompalli — APPROVED** [review]

**Pull request merged**